src/DotLiquid.Tests/FilterTests.cs

```diff
@@ -261,5 +261,21 @@ public void TestContextFilter()
             _context.AddFilters(typeof(ContextFilters));
             Assert.AreEqual(" King Kong has 1000$ ", new Variable("var | bank_statement").Render(_context));
         }
+
+
+        [Test]
+        public void Truncate()

```

Comment:
Is context required to expose the issue with integer conversion? Otherwise there are two existing unit tests under standard filters - TestTruncate and TestTruncateWords

Reply:
The two existing unit tests under standard filters did not fail before the changes in this commit, but this one did. They require the _context and I'm not sure where else they would fit, so I put them in here. Do you have any other suggestions?

Reply:
I see the existing standard filter tests did not fail because they inline the native type input parameter rather than taking it from parsed input / liquid template. This can be done there with lines such as:

Helper.AssertTemplateResult(expected: "Hello", template: "{{ Hello | truncate:4 }}");

(That's what I used just now to reproduce this problem.)

Reply:
Corrected sample unit test:

Helper.AssertTemplateResult(expected: "H...", template: "{{ 'Hello' | truncate:4 }}");

validated with that sample that the generic approach in Strainer could work.

Reply:
I've now moved these unit tests to the existing standard filters unit test methods